### PR TITLE
configure kibana index mode

### PR DESCRIPTION
**Pull request by jcantrill**
This PR allows configuration of the kibana index mode in support of:

* https://trello.com/c/O6MCrGUx/
* https://bugzilla.redhat.com/show_bug.cgi?id=1427254

### CODE REVIEW

**Comment by jcantrill** [comment]
aos-ci-test

**Comment by jcantrill** [comment]
[(3) Ability to seed Kibana dashboard for admins[logging][epic-ois-agl-exp]](https://trello.com/c/O6MCrGUx/454-3-ability-to-seed-kibana-dashboard-for-adminsloggingepic-ois-agl-exp)

File: roles/openshift_logging_elasticsearch/defaults/main.yml [hunk 1]
```diff
@@ -37,6 +37,9 @@ openshift_logging_elasticsearch_storage_group: '65534'
 
 openshift_logging_es_pvc_prefix: "{{ openshift_hosted_logging_elasticsearch_pvc_prefix | default('logging-es') }}"
 
+# config the es plugin to write kibana index based on the index mode
+openshift_logging_elasticsearch_kibana_index_mode: "shared_ops"
```
Comment (jcantrill) [inline roles/openshift_logging_elasticsearch/defaults/main.yml]: @richm @portante Do we want to change the default so ops users go to the same index or is this too much of a surprise and we should require users to opt in?

Comment (richm) [inline roles/openshift_logging_elasticsearch/defaults/main.yml]: @jcantrill good question - open up this question to a wider audience

File: roles/openshift_logging_elasticsearch/tasks/main.yaml [hunk 2]
```diff
@@ -102,6 +102,11 @@
     delete_after: true
 
 # configmap
+- assert:
+    that:
+    - "{{ openshift_logging_elasticsearch_kibana_index_mode in __kibana_index_modes }}"
```
Comment (richm) [inline roles/openshift_logging_elasticsearch/tasks/main.yaml]: do you need to have a string and a template evaulation here?  or can you just use
```
- openshift_logging_elasticsearch_kibana_index_mode in __kibana_index_modes
```
?

Comment (jcantrill) [inline roles/openshift_logging_elasticsearch/tasks/main.yaml]: Nope.  Fixed to remove eval

File: roles/openshift_logging_elasticsearch/defaults/main.yml (hunk 1 above)
Comment (portante) [inline roles/openshift_logging_elasticsearch/defaults/main.yml]: I don't think we can just change the default without migrating dashboards users have already created.  As they'll be lost entirely.

I think it has to be opt-in until we have a story on how to handle such a transition.

**Comment by jcantrill** [comment]
aos-ci-test

**Comment by jcantrill** [comment]
[merge]

**Pull request merged**